__tests__/unit/src/services/bookPriceCalculator.test.js

```diff
@@ -0,0 +1,159 @@
+        .withLowestPrice(19.99)
+        .withSalesRank(1199999)
+        .buildLookup()
+      expect(calculate(book)).toEqual(NOT_INTERESTED_IN_THIS_BOOK)
+
+      const anotherBook = new BookLookupBuilder()
+        .withLowestPrice(19.99)
+        .withSalesRank(900001)
+        .buildLookup()
+      expect(calculate(anotherBook)).toEqual(NOT_INTERESTED_IN_THIS_BOOK)
+    })
+
+    it('salesRank <= 900000 and price < $10', () => {
+      const book = new BookLookupBuilder()
+        .withLowestPrice(9.99)
+        .withSalesRank(900000)
+        .buildLookup()
+      expect(calculate(book)).toEqual(NOT_INTERESTED_IN_THIS_BOOK)
+    })
+  })
+
+  describe('should accept as donation if', () => {
+    it('salesRank > 900000 < 1.2M and price > $20', () => {
+      const book = new BookLookupBuilder()
+        .withLowestPrice(20.01)
+        .withSalesRank(900001)
+        .buildLookup()
+      expect(calculate(book)).toEqual(ACCEPT_AS_DONATION)
+
+      const anotherBook = new BookLookupBuilder()
+        .withLowestPrice(20.01)
+        .withSalesRank(1199999)
+        .buildLookup()
+      expect(calculate(anotherBook)).toEqual(ACCEPT_AS_DONATION)
+    })
+
+    it('salesRank <= 900000 and $20 > price >= $10', () => {
+      const book = new BookLookupBuilder()
+        .withLowestPrice(10.0)
+        .withSalesRank(900000)
+        .buildLookup()
+      expect(calculate(book)).toEqual(ACCEPT_AS_DONATION)
+
+      const anotherBook = new BookLookupBuilder()
+        .withLowestPrice(19.99)
+        .withSalesRank(900000)
+        .buildLookup()
+      expect(calculate(anotherBook)).toEqual(ACCEPT_AS_DONATION)
+    })
+  })
+
+  describe('should calculate offer discount for', () => {
+    it('salesRank <= 199999 and $30 >= price >= $20, 20% PRICE', () => {
+      const pricePercentage = 0.8
+
+      const book = new BookLookupBuilder()
+        .withLowestPrice(20.0)
+        .withSalesRank(199999)
+        .buildLookup()
+      expect(calculate(book)).toEqual(book.price * pricePercentage)
+
+      const anotherBook = new BookLookupBuilder()
+        .withLowestPrice(30.0)
+        .withSalesRank(199999)
+        .buildLookup()
+
+      expect(calculate(anotherBook)).toEqual(anotherBook.price * pricePercentage)
+    })
+
+    it('salesRank <= 199999 and price > $30, 35% PRICE', () => {
+      const pricePercentage = 0.65
+
+      const book = new BookLookupBuilder()
+        .withLowestPrice(30.01)
+        .withSalesRank(199999)
+        .buildLookup()
+      expect(calculate(book)).toEqual(book.price * pricePercentage)
+    })
+
+    it('salesRank <= 199999 and price > $30, 35% PRICE', () => {
+      const pricePercentage = 0.65
+
+      const book = new BookLookupBuilder()
+        .withLowestPrice(30.01)
+        .withSalesRank(199999)
+        .buildLookup()
+      expect(calculate(book)).toEqual(book.price * pricePercentage)
+    })
+
+    it('200 >= salesRank >= 499999 and $20 >= price >= $30, 20% PRICE', () => {
```

Comment:
Pq 200 e não de 1 ???

Reply:
vai perder todos os livros que são de 1 até 200, era 'up to 199K'

Reply:
eu entendo isso por 1 até 199k. @eduardomoroni qq vc acha ?

Reply:
Isso. errei.
Deveria ser 200 000 ou 199 999? lembrando que ta >= e nao >

Reply:
até 200 000 , de 200 001 já entre no outro rate.